client/app/src/main/java/com/example/mrides/Activity/RouteActivity.java

```diff
@@ -33,8 +41,39 @@ protected void onCreate(Bundle savedInstanceState) {
 
     @Override
     public void onClick(View view) {
+        switch(view.getId()) {
+            case R.id.imageTrash:
+                promptUserCancellation();
+                break;
+            default:
+                backToHome();
+                Toast.makeText(RouteActivity.this, "Ratings have been submitted", Toast.LENGTH_SHORT).show();
+                break;
+        }
+
+    }
+
+    public void promptUserCancellation() {
+        new AlertDialog.Builder(this)
```

Comment:
remove this anonymous class. Should be a separate class or an interface implemented by the 
activity.